### PR TITLE
Refactor toolbar

### PR DESCRIPTION
**Pull request by Logende**
Still needs more refactoring but this is at least a start

### CODE REVIEW

File: frontend/config-assistant/src/components/toolbar/TopToolbar.vue [hunk 1]
```diff
@@ -288,7 +285,7 @@ watch(storeToRefs(useSessionStore()).fileData, () => {
         <!-- Wrap the buttons in a div and add margin to it -->
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/toolbar/TopToolbar.vue:285]: All the dialogs should be different components im my oppinion. And they should expose a "show" method so we don't have these weird boolean properties everywhere. 
Btw, do we even use the first Dialog?

File: frontend/config-assistant/src/components/toolbar/fetchSchemaFromUrl.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export async function fetchSchemaFromUrl(schemaURL: string, toast?: any): Promise<void> {
+  try {
+    // Fetch the schema content from the selected schemaURL.
+    const response = await fetch(schemaURL);
+    const schemaContent = await response.json();
+    const schemaName = schemaContent.title || 'Unknown Schema';
+    useSessionStore().lastChangeResponsible = ChangeResponsible.Menubar;
+    // Update the schemaData in the dataStore with the fetched schema content.
+    useDataStore().schemaData = schemaContent;
+    // Always clear the data without prompting the user.
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/toolbar/fetchSchemaFromUrl.ts:17]: This comment is wrong

File: frontend/config-assistant/src/components/toolbar/fetchSchemaFromUrl.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import {newEmptyFile} from '@/components/toolbar/clearFile';
+
+export async function fetchSchemaFromUrl(schemaURL: string, toast?: any): Promise<void> {
+  try {
+    // Fetch the schema content from the selected schemaURL.
+    const response = await fetch(schemaURL);
+    const schemaContent = await response.json();
+    const schemaName = schemaContent.title || 'Unknown Schema';
+    useSessionStore().lastChangeResponsible = ChangeResponsible.Menubar;
+    // Update the schemaData in the dataStore with the fetched schema content.
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/toolbar/fetchSchemaFromUrl.ts:15]: This comment is not very helpful

File: frontend/config-assistant/src/components/toolbar/fetchSchemaFromUrl.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      toast.add({
+        severity: 'info',
+        summary: 'Info',
+        detail: `"${schemaName}" fetched successfully!`,
+        life: 3000,
+      });
+    }
+  } catch (error) {
+    // Handle the error if there's an issue fetching the schema.
+    errorService.onError(error);
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/toolbar/fetchSchemaFromUrl.ts:30]: the try catch is not necessary, if you call errorService.onError(error);
This is the default error handler anyway, so you can just remove the try catch

File: frontend/config-assistant/src/components/toolbar/fetchWebSchemas.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      fetchedSchemas.push({
+        label: schema.name,
+        icon: 'pi pi-fw pi-code',
+        url: schema.url,
+        key: schema.key,
+      });
+    });
+  } catch (error) {
+    // Handle the error if there's an issue fetching the schema.
+    errorService.onError(error);
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/toolbar/fetchWebSchemas.ts:24]: same here

**Review by PaulBredl — COMMENTED** [review]
We can merge this as a start, but there is still a lot to do in my oppinion

**Comment by Logende** [comment]
> We can merge this as a start, but there is still a lot to do in my oppinion

Yeah there for sure still is a lot to do and the code is a mess still, but this refactoring already does some improvements. Doing all at once would be too overwhelming.

**Comment by Logende** [comment]
I will make some more adaptions based on your comments and then I would merge

**Comment by Logende** [comment]
@PaulBredl okay, I would merge now. Also to avoid more merge conflicts. More refactoring can be done in a follow-up step.

**Review by PaulBredl — APPROVED** [review]

**Pull request merged**